wallet/core/src/storage/local/interface.rs

```diff
@@ -24,25 +29,38 @@ pub fn make_filename(title: &Option<String>, filename: &Option<String>) -> Strin
     } else if let Some(title) = title {
         slugify!(title)
     } else {
-        super::DEFAULT_WALLET_FILE.to_string()
+        super::default_wallet_file().to_string()
     }
 }
 
+#[derive(Clone)]
 pub enum Store {
     Resident,
     Storage(Storage),
 }
 
+impl Store {
+    fn filename(&self) -> Option<String> {
+        match self {
+            Store::Resident => None,
+            Store::Storage(storage) => Some(storage.filename_as_string()),
+        }
+    }
+
+    pub fn is_resident(&self) -> bool {
+        matches!(self, Store::Resident)
+    }
+}
+
 pub(crate) struct LocalStoreInner {
     pub cache: Arc<Mutex<Cache>>,
-    pub store: Store,
+    pub store: Mutex<Arc<Store>>,
```

Comment:
rwlock?

Reply:
Switched to RwLock.  It's actually not the `store` that needed that but `cache`.  Both have been updated:  https://github.com/aspectron/rusty-kaspa/commit/5ea869650c076b8dec9f7f48645d8888129c079d